changelog.d/12304.misc

```diff
@@ -0,0 +1 @@
+Refactor `create_new_client_event` to use a new parameter, `state_event_ids`, which accurately describes the usage with [MSC2716](https://github.com/matrix-org/matrix-doc/pull/2716) instead of abusing `auth_event_ids`.
```

Comment:
@erikjohnston Request for you to merge. The CI failure seems to be unrelated and also on the latest [`develop`](https://github.com/matrix-org/synapse/runs/5725400049). The [CI was passing](https://github.com/matrix-org/synapse/runs/5700408377) on the commit a few days ago before adding the changelog suggestion.